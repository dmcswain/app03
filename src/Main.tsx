import { useMemo } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { CssBaseline, ThemeProvider, useMediaQuery } from '@mui/material';
import getTheme from './theme';
import StoreProvier from 'store/Provider';

interface Props {
   children: React.ReactNode;
}

const Main: React.FC<Props> = ({ children }) => {
   const isDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
   const theme = useMemo(() => getTheme(isDarkMode), [isDarkMode]);

   return (
      <ThemeProvider theme={theme}>
         <BrowserRouter>
            <StoreProvier>
               <CssBaseline />
               {children}
            </StoreProvier>
         </BrowserRouter>
      </ThemeProvider>
   );
};

export default Main;
