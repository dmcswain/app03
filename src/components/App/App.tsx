import { Route, Routes, useLocation } from 'react-router-dom';
import LoginDialog from 'components/LoginDialog';
import Layout from 'components/Layout';
import Home from 'pages/Home';

function App() {
   const location = useLocation();
   const state = location.state as { backgroundLocation?: Location };

   return (
      <>
         <Routes location={state?.backgroundLocation || location}>
            <Route path='/' element={<Layout />}>
               <Route index element={<Home />} />
               <Route path='login' element={<LoginDialog />} />
            </Route>
         </Routes>
         {state?.backgroundLocation && (
            <Routes>
               <Route path='login' element={<LoginDialog />} />
            </Routes>
         )}
      </>
   );
}

export default App;
