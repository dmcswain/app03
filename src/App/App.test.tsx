import { render } from '@testing-library/react';
import Main from 'Main';
import App from './App';

test('renders app', () => {
   const wrapper = ({ children }: { children: React.ReactNode }) => (
      <Main>{children}</Main>
   );

   const { asFragment } = render(<App />, { wrapper });

   expect(asFragment()).toMatchSnapshot();
});
