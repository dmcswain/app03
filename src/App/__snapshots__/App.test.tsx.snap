// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-9c3o97-MuiToolbar-root"
    >
      <a
        href="/"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
        >
          <img
            alt="logo"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="logo192.png"
          />
        </div>
      </a>
      <div
        class="user"
      >
        <a
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-sbfmij-MuiButtonBase-root-MuiButton-root"
          href="/login"
          tabindex="0"
        >
          Login
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </header>
  <main
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    />
    <div
      class="MuiBox-root css-ovbz0g"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
      >
        Home
      </h2>
    </div>
  </main>
  ;
</DocumentFragment>
`;
